test/extended/util/controllersmanager.go

```diff
@@ -0,0 +1,323 @@
+package util
+
+import (
+	"container/list"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	etcdclient "github.com/coreos/go-etcd/etcd"
+	g "github.com/onsi/ginkgo"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api/latest"
+)
+
+const (
+	EtcdLeasePath = `/openshift.io/leases/controllers`
+)
+
+// ControllersManager allows to start and stop controllers instances,
+// distinguish currently active one by looking up leaseID in etcd storage and
+// allows for updating and deleting of the lease.
+type ControllersManager struct {
```

Comment:
A manager for our managers (controllers) concerns me.  I'll keep reading, but I expected this test to be building and scaling a replication controller for our controllers.  Maybe create a bare pod to prime the system, but then an RC to test scaling out.